pygfx/objects/_skins.py

```diff
@@ -0,0 +1,206 @@
+
+    def pose(self):
+        """Reset the skeleton to the binding-time pose."""
+        for i, bone in enumerate(self.bones):
+            bone.world.matrix = np.linalg.inv(self.bone_inverses[i])
+
+        for bone in self.bones:
+            if bone.parent and isinstance(bone.parent, Bone):
+                bone.local.matrix = (
+                    np.linalg.inv(bone.parent.world.matrix) @ bone.world.matrix
+                )
+            else:
+                bone.local.matrix = bone.world.matrix
+
+            bone.local.position, bone.local.rotation, bone.local.scale = (
+                la.mat_decompose(bone.local.matrix)
+            )
+
+    def update(self):
+        # TODO: we should update the bone matrices buffer automatically by some mechanism.
+        # See: https://github.com/pygfx/pygfx/pull/715#issuecomment-2046493145
+        """Update the bone matrices buffer."""
+
+        for i, bone in enumerate(self.bones):
+            self.bone_matrices_buffer.data[i]["bone_matrices"] = (
+                bone.world.matrix @ self.bone_inverses[i]
+            ).T
+
+        self.bone_matrices_buffer.update_range()
+
+    def get_bone(self, name):
+        for bone in self.bones:
+            if bone.name == name:
+                return bone
+        return None
+
+
+class SkinnedMesh(Mesh):
+    """A skinned mesh.
+
+    A mesh that has a Skeleton with bones that can then be used to animate the vertices of the geometry.
+
+    """
+
+    uniform_type = dict(
+        Mesh.uniform_type,
+        bind_matrix="4x4xf4",
+        bind_matrix_inv="4x4xf4",
+    )
+
+    def __init__(self, geometry, material):
+        super().__init__(geometry, material)
+
+        self.bind_matrix = np.eye(4)
+        self.bind_matrix_inv = np.eye(4)
+
+        self._bind_mode = BindMode.attached
+
+    @property
+    def bind_matrix(self):
+        """The base matrix that is used for the bound bone transforms."""
+        return self._bind_matrix
+
+    @bind_matrix.setter
+    def bind_matrix(self, value):
+        self._bind_matrix = value
+        self.uniform_buffer.data["bind_matrix"] = self._bind_matrix.T
+        self.uniform_buffer.update_range()
+
+    @property
+    def bind_matrix_inv(self):
+        """The base matrix that is used for resetting the bound bone transforms."""
+        return self._bind_matrix_inv
+
+    @bind_matrix_inv.setter
+    def bind_matrix_inv(self, value):
+        self._bind_matrix_inv = value
+        self.uniform_buffer.data["bind_matrix_inv"] = self._bind_matrix_inv.T
+        self.uniform_buffer.update_range()
+
+    @property
+    def bind_mode(self):
+        """
+        How a skinned mesh is bound to its skeleton. Either "attached" or "detached".
+
+        "attached" means the skinned mesh shares the same world space as the skeleton.
+
+        In contrast, "detached" is useful when sharing a skeleton across multiple skinned meshes.
+        Default is "attached".
```

Comment:
We can just refer to the docs of the enum.
```suggestion
        See :obj:`pygfx.utils.enums.BindMode`:. Default "attached".
```